graphql/resolvers.go

```diff
@@ -2530,6 +2524,129 @@ func (r *queryResolver) BbGetCreatedTickets(ctx context.Context, taskID string)
+		if err != nil {
+			return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error building APIVersion from service: %s", err.Error()))
+		}
+
+		// If we try to access a version that does not have the Activated flag we must manually verify it
+		// After we verify if a version is activated we cache that field so subsequent queries are faster.
+		if v.Activated == nil {
+			defaultSort := []task.TasksSortOrder{
+				{Key: task.DisplayNameKey, Order: 1},
+			}
+			opts := data.TaskFilterOptions{
+				Statuses:        []string{},
+				BaseStatuses:    []string{},
+				Variants:        []string{},
+				TaskNames:       []string{},
+				Page:            0,
+				Limit:           0,
+				FieldsToProject: []string{},
+				Sorts:           defaultSort,
+			}
+			tasks, _, err := r.sc.FindTasksByVersion(v.Id, opts)
+			if err != nil {
+				return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error fetching tasks for version %s : %s", v.Id, err.Error()))
+			}
+			if !task.AnyActiveTasks(tasks) {
+				err = v.SetNotActivated()
+				if err != nil {
+					return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error updating version activated status for %s, %s", v.Id, err.Error()))
+				}
+			} else {
+				err = v.SetActivated()
+				if err != nil {
+					return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error updating version activated status for %s, %s", v.Id, err.Error()))
+				}
+			}
+		}
+		mainlineCommitVersion := MainlineCommitVersion{}
+
+		if utility.FromBoolPtr(v.Activated) {
+			activatedVersionCount++
+			mainlineCommits.NextPageOrderNumber = &v.RevisionOrderNumber
+			mainlineCommitVersion.Version = &apiVersion
+			mainlineCommits.Versions = append(mainlineCommits.Versions, &mainlineCommitVersion)
+		} else {
+			if len(mainlineCommits.Versions) > 0 {
+				lastMainlineCommit := mainlineCommits.Versions[len(mainlineCommits.Versions)-1]
+				if lastMainlineCommit.RolledUpVersions != nil {
+					lastMainlineCommit.RolledUpVersions = append(lastMainlineCommit.RolledUpVersions, &apiVersion)
+				}
+			} else {
+				mainlineCommitVersion.RolledUpVersions = []*restModel.APIVersion{&apiVersion}
+				mainlineCommits.Versions = append(mainlineCommits.Versions, &mainlineCommitVersion)
+			}
+
+		}
+
+	}
+
+	return &mainlineCommits, nil
+}
+
+type versionResolver struct{ *Resolver }
+
+// Returns grouped build variants for a version. Will not return build variants for unactivated versions
+func (r *versionResolver) BuildVariants(ctx context.Context, v *restModel.APIVersion, options *BuildVariantOptions) ([]*GroupedBuildVariant, error) {
+	if !utility.FromBoolPtr(v.Activated) {
+		return nil, nil
+	}
+	variants := []string{}
+	tasks := []string{}
+	statuses := []string{}
+	if options.Variants != nil {
+		variants = options.Variants
+	}
+	if options.Tasks != nil {
+		tasks = options.Tasks
+	}
+	if options.Statuses != nil {
+		statuses = options.Statuses
+	}
+	return generateBuildVariants(ctx, r.sc, *v.Id, variants, tasks, statuses)
```

Comment:
You can just pass in each option directly; there's no reason we can't pass in a nil list.